zookeeper-client/zookeeper-client-c/src/cli.c

```diff
@@ -876,18 +929,43 @@ int main(int argc, char **argv) {
 #endif
     zoo_deterministic_conn_order(1); // enable deterministic order
 
-#ifdef HAVE_OPENSSL_H
-    if (!cert) {
-        zh = zookeeper_init(hostPort, watcher, 30000, &myid, NULL, flags);
-    } else {
-        zh = zookeeper_init_ssl(hostPort, cert, watcher, 30000, &myid, NULL, flags);
+#ifdef HAVE_CYRUS_SASL_H
+    if (mechlist) {
+        zoo_sasl_params_t sasl_params = { 0 };
+
+        if (sasl_client_init(NULL) != SASL_OK) {
+            return 1;
```

Comment:
This is the status code which is returned to the OS.  `cli.c` returns a mixture of 1, 2 and sometimes `errno` (?!) on error.  It would be good to rationalize this, but I would suggest doing it in one go (and perhaps just using `EXIT_FAILURE`).

I have added some diagnostics, however, as there is no good reason to "just exit."